thrust/system/detail/generic/shuffle.inl

```diff
@@ -0,0 +1,184 @@
+/*
+ *  Copyright 2008-2013 NVIDIA Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include <thrust/detail/config.h>
+#include <thrust/detail/temporary_array.h>
+#include <thrust/iterator/discard_iterator.h>
+#include <thrust/iterator/transform_iterator.h>
+#include <thrust/scan.h>
+#include <thrust/iterator/transform_output_iterator.h>
+#include <thrust/random/linear_feedback_shift_engine.h>
+#include <thrust/system/detail/generic/shuffle.h>
+
+namespace thrust {
+namespace system {
+namespace detail {
+namespace generic {
+
+class FeistelBijectiveFunction {
```

Comment:
Bit of a nitpick, but Thrust uses `snake_case` and reserves `CamelCase` for the names of template parameters.